service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/CaseManagementOrderReadyForJudgeReviewEventHandler.java

```diff
@@ -45,16 +46,19 @@ public void sendEmailForCaseManagementOrderReadyForJudgeReviewToAllocatedJudge(
         final CaseManagementOrderReadyForJudgeReviewEvent event) {
         EventData eventData = new EventData(event);
         CaseData caseData = mapper.convertValue(eventData.getCaseDetails().getData(), CaseData.class);
-        if (featureToggleService.isAllocatedJudgeNotificationEnabled(CMO)) {
-            if (caseData.allocatedJudgeExists()) {
-                AllocatedJudgeTemplateForCMO parameters = caseManagementOrderEmailContentProvider
-                    .buildCMOReadyForJudgeReviewNotificationParameters(eventData.getCaseDetails());
 
-                String email = caseData.getAllocatedJudge().getJudgeEmailAddress();
+        if (featureToggleService.isAllocatedJudgeNotificationEnabled(CMO) && hasAllocatedJudgeEmail(caseData)) {
+            AllocatedJudgeTemplateForCMO parameters = caseManagementOrderEmailContentProvider
+                .buildCMOReadyForJudgeReviewNotificationParameters(eventData.getCaseDetails());
 
-                notificationService.sendEmail(CMO_READY_FOR_JUDGE_REVIEW_NOTIFICATION_TEMPLATE_JUDGE, email, parameters,
-                    eventData.getReference());
-            }
+            String email = caseData.getAllocatedJudge().getJudgeEmailAddress();
+
+            notificationService.sendEmail(CMO_READY_FOR_JUDGE_REVIEW_NOTIFICATION_TEMPLATE_JUDGE, email, parameters,
+                eventData.getReference());
         }
     }
+
+    private boolean hasAllocatedJudgeEmail(CaseData caseData) {
+        return caseData.allocatedJudgeExists() && isNotEmpty(caseData.getAllocatedJudge().getJudgeEmailAddress());
+    }
```

Comment:
This method is essentially the same as the one in c2 upload handler, is it worth moving to case data or what you think?